pkgs/top-level/all-packages.nix

```diff
@@ -32416,8 +32416,16 @@ with pkgs;
   ifstat-legacy = callPackage ../tools/networking/ifstat-legacy { };
 
   isabelle = callPackage ../applications/science/logic/isabelle {
-    polyml = lib.overrideDerivation polyml (_: {
+    polyml = polyml.overrideAttrs (_: {
       configureFlags = [ "--enable-intinf-as-int" "--with-gmp" "--disable-shared" ];
+      buildFlags = [ "compiler" ];
+      version = "for-isabelle-2021-1";
+      src = fetchFromGitHub {
```

Comment:
Any particular reason to lock the polyml version here?

Reply:
I want to use a version that matches the upstream distribution as closely as possible just in case Isabelle relies on it. I know that Isabelle relies on a particular version of Z3, and using the version from nixpkgs has broken things before (#99399).

Reply:
I see